Incompatible with `async-trait` crate

### Code tested:

```rust
// async_trait_support.rs
extern crate async_trait;
use extend::ext;

#[async_trait::async_trait]
#[ext]
impl i32 {
    async fn bar() {}
}

fn main() {}
```
### Result:

```
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0706]: functions in traits cannot be declared `async`
 --> $DIR/async_trait_support.rs:4:1
  |
4 | #[ext]
  | ^^^^^^ `async` because of this
  |
  = note: `async` trait functions are not currently supported
  = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait

error[E0195]: lifetime parameters or bounds on method `bar` do not match the trait declaration
 --> $DIR/async_trait_support.rs:4:7
  |
4 | #[ext]
  | ------^ lifetimes do not match method in trait
  | |
  | lifetimes in impl do not match this method in trait
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
```

### Expected result:
Successful compile, produces trait as usual that can be used with `async_trait`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,11 +6,11 @@ for Rust libraries in [RFC #1105](https://github.com/rust-lang/rfcs/blob/master/
 
 ## Unreleased
 
-None.
+- Add async-trait compatibility.
 
 ### Breaking changes
 
-None.
+- Other attributes put on the `impl` would previously only be included on the generated trait. They're now included on both the trait and the implementation.
 
 ## 0.2.1 - 2020-08-29
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,6 +20,7 @@ proc-macro-error = "1"
 
 [dev_dependencies]
 trybuild = "1.0.17"
+async-trait = "0.1.40"
 
 [lib]
 proc-macro = true

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -106,6 +106,35 @@
 //! }
 //! ```
 //!
+//! # async-trait compatibility
+//!
+//! Async extensions are supported via [async-trait](https://crates.io/crates/async-trait).
+//!
+//! Be aware that you need to add `#[async_trait]` _below_ `#[ext]`. Otherwise the `ext` macro
+//! cannot see the `#[async_trait]` attribute and pass it along in the generated code.
+//!
+//! Example:
+//!
+//! ```
+//! use extend::ext;
+//! use async_trait::async_trait;
+//!
+//! #[ext]
+//! #[async_trait]
+//! impl String {
+//!     async fn read_file() -> String {
+//!         // ...
+//!         # todo!()
+//!     }
+//! }
+//! ```
+//!
+//! # Other attributes
+//!
+//! Other attributes provided _below_ `#[ext]` will be passed along to both the generated trait and
+//! the implementation. See [async-trait compatibility](#async-trait-compatibility) above for an
+//! example.
+//!
 //! [extension traits]: https://dev.to/matsimitsu/extending-existing-functionality-in-rust-with-traits-in-rust-3622
 
 #![doc(html_root_url = "https://docs.rs/extend/0.2.1")]
@@ -210,6 +239,7 @@ pub fn ext(
             )*
         }
 
+        #(#attrs)*
         impl #impl_generics #ext_trait_name #ty_generics for #self_ty #where_clause {
             #(#items)*
         }

--- a/tests/compile_pass/async_trait.rs
+++ b/tests/compile_pass/async_trait.rs
@@ -1,0 +1,12 @@
+use extend::ext;
+use async_trait::async_trait;
+
+#[ext]
+#[async_trait]
+impl String {
+    async fn foo() -> usize {
+        1
+    }
+}
+
+fn main() {}

